app/Http/Controllers/Admin/RoleController.php

```diff
@@ -0,0 +1,336 @@
+            ]);
+        } catch (Exception $e) {
+            DB::rollback();
+
+            return response()->json([
+                'error' => true,
+                'message' => 'Fail !',
+            ]);
+        }
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, $id)
+    {
+        $data = $request->all();
+
+        DB::beginTransaction();
+
+        try {
+            $rules = [
+                'name' => 'required|string',
+                'display_name' => 'required|string',
+            ];
+
+            $messages = [
+                'name.required' => __('Role') . __('Required'),
+                'name.string' => __('Role') . __('String'),
+                'display_name.required' => __('Display name') . __('Required'),
+                'display_name.string' => __('Display name') . __('String'),
+            ];
+
+            $validator = Validator::make($data, $rules, $messages);
+
+            if ($validator->fails()) {
+                return response()->json([
+                    'error' => 'valid',
+                    'message' => $validator->errors(),
+                ]);
+            } else {
+                $this->roleRepository->update($data, $id);
+
+                DB::commit();
+
+                return response()->json([
+                    'error' => false,
+                    'message' => 'Success !',
+                ]);
+            }
+        } catch (Exception $e) {
+            return response()->json([
+                'error' => true,
+                'message' => 'Fail !',
+            ]);
+        }
+    }
+
+    /**
+     * Remove the specified resource from storage.
+     *
+     * @param  int  $id
+     * @return \Illuminate\Http\Response
+     */
+    public function destroy($id)
+    {
+        DB::beginTransaction();
+
+        try {
+            DB::table('roles')->where('id', $id)->delete();
+
+            DB::commit();
+
+            return response()->json([
+                'error' => false,
+                'message' => 'Success !',
+            ]);
+        } catch (Exception $e) {
+            DB::rollback();
+
+            return response()->json([
+                'error' => true,
+                'message' => 'Fail !',
+            ]);
+        }
+    }
+
+    /**
+     * [getRoles : Lấy ra danh sách vai trò]
+     * @return [type] [description]
+     */
+    public function getRoles()
+    {
+        $roles = $this->roleRepository->orderBy('id', 'desc')->get();
+
+        return Datatables::of($roles)
+            ->addIndexColumn()
+
+            ->editColumn('created_at', function ($role){
+                $time = $role->created_at;
+
+                $time_numb = strtotime($time);
+
+                return date('H:i | d-m-Y', $time_numb);
+            })
```

Comment:
**Reporter**: CHECKSTYLE</br>**Rule**: Framgia.Variables.ValidVariableName.NotCamelCaps</br>**Severity**: error</br>**File**: app/Http/Controllers/Admin/RoleController.php</br>Variable "time_numb" is not in valid camel caps format